.githooks/pre-commit

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+files=()
+for file in `git diff --cached --name-only --diff-filter=ACMRT | grep -E "\.(cpp|hpp|c|h)$"`; do
+  if ! cmp -s <(git show :${file}) <(git show :${file}|clang-format); then
+    files+=("${file}")
```

Comment:
Somewhere here or in the grep on line 4 there should be a filter to exclude source files in two subfolders:
- `interfaces` to avoid formatting/interface file generation cycles
- `ThirdParty` to make sure we aren't messing up the formatting of some vendored code that doesn't follow the same style

Reply:
I have added filters for these two directories.